test/tests/services/fields-flattener.test.js

```diff
@@ -0,0 +1,197 @@
+
+        jest.resetAllMocks();
+        const fieldsFlattener = new FieldsFlattener({}, { });
+        fieldsFlattener.validateOptions();
+
+        expect(errorLoggerSpy).toHaveBeenCalledTimes(1);
+      });
+    });
+
+    describe('when the field provided is invalid', () => {
+      describe('when the field does not exist', () => {
+        describe('when the field is of type string', () => {
+          it('should display an warning message', () => {
+            expect.assertions(1);
+
+            jest.resetAllMocks();
+            const fieldsFlattener = new FieldsFlattener({ fields: [] }, ['test']);
+            fieldsFlattener.validateOptions();
+
+            expect(warnLoggerSpy).toHaveBeenCalledTimes(1);
+          });
+
+          it('should remove the non existing field from fields to flatten', () => {
+            expect.assertions(1);
+
+            const fieldsFlattener = new FieldsFlattener({ fields: [] }, ['test']);
+            fieldsFlattener.validateOptions();
+
+            expect(fieldsFlattener.flatten).toHaveLength(0);
+          });
+        });
+
+        describe('when the field is of type object', () => {
+          it('should display a warning message', () => {
+            expect.assertions(1);
+
+            jest.resetAllMocks();
+            const fieldsFlattener = new FieldsFlattener({ fields: [] }, [{ field: 'test' }]);
+            fieldsFlattener.validateOptions();
+
+            expect(warnLoggerSpy).toHaveBeenCalledTimes(1);
+          });
+
+          it('should remove the non existing field from fields to flatten', () => {
+            expect.assertions(1);
+
+            const fieldsFlattener = new FieldsFlattener({ fields: [] }, [{ field: 'test' }]);
+            fieldsFlattener.validateOptions();
+
+            expect(fieldsFlattener.flatten).toHaveLength(0);
+          });
+        });
+      });
+
+      describe('when a field has not been specified in flatten object', () => {
+        it('should display a warning message', () => {
+          expect.assertions(1);
+
+          jest.resetAllMocks();
+          const fieldsFlattener = new FieldsFlattener({ fields: [] }, [{ }]);
+          fieldsFlattener.validateOptions();
+
+          expect(warnLoggerSpy).toHaveBeenCalledTimes(1);
+        });
+
+        it('should remove the non existing field from fields to flatten', () => {
+          expect.assertions(1);
+
+          const fieldsFlattener = new FieldsFlattener({ fields: [] }, [{ }]);
+          fieldsFlattener.validateOptions();
+
+          expect(fieldsFlattener.flatten).toHaveLength(0);
+        });
+      });
+    });
+  });
+
+  describe('flattening a field', () => {
+    it('should merge fields name with | as separator', () => {
+      expect.assertions(1);
+
+      const schema = generateEngineSchema();
+
+      const fieldsFlattener = new FieldsFlattener(schema, ['engine']);
+      fieldsFlattener.flattenFields();
+
+      expect(schema.fields[0].field).toContain('|');
```

Comment:
would be nice to validate the whole field name

Reply:
👍